### PR TITLE
Rename ResultSetMapper to ResultSetScanner

### LINKED ISSUE #889: Clarify `ResultSetMapper` -- does it differ from v2?  who is responsible for advancing the result set?
per @mikebell90 the `ResultSetMapper` contract is confusing and may be inconsistent with the v2 contract -- it looks like we assume the mapper will do row advancing and the like itself.  At the very least this change needs documentation, and we should think about whether this is the right behavior.

### PR DESCRIPTION
**Pull request by stevenschlansker**
The name 'mapper' is already used for a different concept

cc @mikebell90

Fixes #889

### CODE REVIEW

File: RELEASE_NOTES [hunk 1]
```diff
@@ -1,4 +1,7 @@
-3.0.0-beta3 (in development)
+3.0.0-beta4 (in development)
+  - [breaking] ResultSetMapper -> ResultSetScanner; reducing overloaded 'Mapper'
+
+3.0.0-beta
```
Comment (qualidafial) [inline RELEASE_NOTES]: Should be beta3

File: docs/src/adoc/index.adoc [hunk 2]
```diff
@@ -909,6 +909,19 @@ single remains, and then return that.
 
 TODO: example
 
+===== ResultSetScanner
+
+The *ResultSetScanner* interface accepts a lazy-provided *ResultSet*
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: Nit: lazily

File: docs/src/adoc/index.adoc [hunk 3]
```diff
@@ -909,6 +909,19 @@ single remains, and then return that.
...
+and produces the result Jdbi returns from statement execution.
+
+Most of the above operations are implemented in terms of *ResultSetScanner*.
+The Scanner has ownership of the ResultSet and may advance or seek it.
+
+The return value ends up being the final result of statement execution.
+
+Most uses should prefer using the higher level result collectors described above,
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: Did you mean users?

File: docs/src/adoc/index.adoc [hunk 4]
```diff
@@ -909,6 +909,19 @@ single remains, and then return that.
...
+but someone's gotta do the dirty work.
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: :laughing:

**Comment by mikebell90** [comment]
I agree. That's a good name thanks guys

On Sep 26, 2017, at 8:06 PM, Steven Schlansker <notifications@github.com<mailto:notifications@github.com>> wrote:


The name 'mapper' is already used for a different concept

cc @mikebell90<https://github.com/mikebell90>

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/jdbi/jdbi/pull/891

Commit Summary

  *   Rename ResultSetMapper to ResultSetScanner

File Changes

  *   M RELEASE_NOTES<https://github.com/jdbi/jdbi/pull/891/files#diff-0> (5)
  *   M core/src/main/java/org/jdbi/v3/core/result/ResultBearing.java<https://github.com/jdbi/jdbi/pull/891/files#diff-1> (16)
  *   R core/src/main/java/org/jdbi/v3/core/result/ResultSetScanner.java<https://github.com/jdbi/jdbi/pull/891/files#diff-2> (17)
  *   M core/src/main/java/org/jdbi/v3/core/statement/PreparedBatch.java<https://github.com/jdbi/jdbi/pull/891/files#diff-3> (6)
  *   M core/src/main/java/org/jdbi/v3/core/statement/Query.java<https://github.com/jdbi/jdbi/pull/891/files#diff-4> (6)
  *   M docs/src/adoc/index.adoc<https://github.com/jdbi/jdbi/pull/891/files#diff-5> (10)

Patch Links:

  *   https://github.com/jdbi/jdbi/pull/891.patch
  *   https://github.com/jdbi/jdbi/pull/891.diff

**Pull request merged**